xknx/devices/fan.py

```diff
@@ -93,23 +108,54 @@ def __init__(
 
     def _iter_remote_values(self) -> Iterator[RemoteValue[Any, Any]]:
         """Iterate the devices RemoteValue classes."""
-        yield from (self.speed, self.oscillation)
+        yield from (self.switch, self.speed, self.oscillation)
 
     @property
     def supports_oscillation(self) -> bool:
         """Return if fan supports oscillation."""
         return self.oscillation.initialized
 
-    async def set_speed(self, speed: int) -> None:
-        """Set the fan to a desginated speed."""
-        await self.speed.set(speed)
+    @property
+    def is_on(self) -> bool:
+        """Return the current fan state of the device."""
+        if self.switch.initialized:
+            return bool(self.switch.value)
+        return bool(self.current_speed)
+
+    async def turn_on(self, speed: int | None = None) -> None:
+        """Turn on fan."""
+        if self.switch.initialized:
+            await self.set_on()
+        else:
+            await self.set_speed(speed)
+
+    async def turn_off(self) -> None:
+        """Turn off fan."""
+        if self.switch.initialized:
+            await self.set_off()
+        else:
+            await self.set_speed(0)
+
+    async def set_on(self) -> None:
+        """Switch on fan."""
+        await self.switch.on()
+
+    async def set_off(self) -> None:
+        """Switch off fan."""
+        await self.switch.off()
```

Comment:
I guess we don't need set_on and turn_on methods doing essentially the same.
```suggestion
    async def turn_on(self, speed: int | None = None) -> None:
        """Turn on fan."""
        if self.switch.initialized:
            await self.switch.on()
        else:
            await self.set_speed(speed)

    async def turn_off(self) -> None:
        """Turn off fan."""
        if self.switch.initialized:
            await self.switch.off()
        else:
            await self.set_speed(0)
```